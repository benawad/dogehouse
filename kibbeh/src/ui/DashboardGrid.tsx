import React from "react";

interface DashboardGridProps {}

export const DashboardInnerGrid: React.FC<DashboardGridProps> = ({
  children,
}) => {
  return (
    <div
      style={{
        display: "grid",
        gridTemplateColumns: "235px 600px 325px",
        columnGap: 60,
      }}
    >
      {children}
    </div>
  );
};

export const DashboardGrid: React.FC<DashboardGridProps> = ({ children }) => {
  return (
    <div className={`flex justify-center w-full min-h-screen bg-primary-900`}>
      <DashboardInnerGrid>{children}</DashboardInnerGrid>
    </div>
  );
};
