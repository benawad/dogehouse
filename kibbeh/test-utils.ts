import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
