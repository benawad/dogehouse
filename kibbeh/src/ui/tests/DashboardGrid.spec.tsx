import React from "react";

import { render } from "../../../test-utils";
import { DashboardGrid } from "../DashboardGrid";

describe("DashboardGrid", () => {
  it("should render correctly", () => {
    const { getByTestId, getByText } = render(
      <DashboardGrid>Child</DashboardGrid>
    );
    const component = getByTestId("dashboard-grid");
    const child = getByText("Child");

    expect(component).toBeVisible();
    expect(child).toBeVisible();
  });

  it("should match snapshot", () => {
    const { getByTestId } = render(<DashboardGrid>Child</DashboardGrid>);
    const component = getByTestId("dashboard-grid");

    expect(component).toMatchSnapshot();
  });
});
