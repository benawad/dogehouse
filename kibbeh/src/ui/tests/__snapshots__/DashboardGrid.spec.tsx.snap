// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardGrid should match snapshot 1`] = `
<div
  class="flex justify-center w-full min-h-screen bg-primary-900"
  data-testid="dashboard-grid"
>
  <div
    style="display: grid; grid-template-columns: 235px 600px 325px; column-gap: 60px;"
  >
    Child
  </div>
</div>
`;
